src/op_analytics/cli/subcommands/pulls/agora/delegate_events.py

```diff
@@ -0,0 +1,170 @@
+from dataclasses import dataclass
+from op_analytics.cli.subcommands.pulls.agora.data_access import (
+    Agora,
+    write,
+    _camelcase_to_snakecase,
+    parse_isoformat_with_z,
+)
+from op_analytics.coreutils.logger import structlog
+from op_analytics.coreutils.request import get_data
+from op_analytics.coreutils.threads import run_concurrently
+
+# from op_analytics.coreutils.time import parse_isoformat_with_tz
+from typing import Any
+import requests
+from tenacity import retry, stop_after_attempt, wait_exponential, retry_if_exception_type
+import requests.exceptions
+import pandas as pd
+import os
+
+log = structlog.get_logger()
+
+BASE_URL = "https://vote.optimism.io/api/v1"
+DELEGATES_ENDPOINT = f"{BASE_URL}/delegates"
+API_KEY = os.environ["AGORA_API_KEY"]
+session = requests.Session()  # Create a session object
+
+
+@dataclass
+class AgoraDelegateVotes:
+    votes_df: pd.DataFrame
+
+
+@dataclass
+class PaginatedResponse:
+    has_next: bool
+    next_offset: int
+    data: Any
+
+
+@dataclass
+class SimplePaginator:
+    url: str
+    limit: int = 50
+
+    @retry(
+        stop=stop_after_attempt(3),
+        wait=wait_exponential(multiplier=1, min=10, max=40),
+        retry=retry_if_exception_type(
+            (requests.exceptions.Timeout, requests.exceptions.ConnectionError)
```

Comment:
The requests session that we use already mounts a Retry adapter. We can set the number of connect error retries there. Right now we don't specify it so it deafults to 0.

Reply:
Used the proper functionality from request.py for this